crates/vimfuncs/build.rs

```diff
@@ -26,160 +28,158 @@ pub struct IfDef {
     pub proc_else: Box<FuncToken>,
 }
 
-impl FuncToken {
-    fn string(self) -> String {
-        match self {
-            FuncToken::String(s) => s,
-            _ => unreachable!("{:?}", self),
-        }
-    }
-
-    fn number(self) -> usize {
-        match self {
-            FuncToken::Number(s) => s,
-            _ => unreachable!("{:?}", self),
-        }
-    }
-
-    fn identifier(self) -> String {
-        match self {
-            FuncToken::Identifier(s) => s,
-            _ => unreachable!("{:?}", self),
-        }
-    }
-}
-
-impl FuncLexer {
-    pub fn new(input: &str) -> Self {
+impl<'a> FuncLexer<'a> {
+    pub fn new(input: &'a str) -> Self {
         Self {
-            position: 0,
-            chars: input.chars().collect(),
+            chars: input.chars().peekable(),
         }
     }
 
     pub fn lex(&mut self) -> Vec<FuncToken> {
         let mut result = vec![];
 
-        loop {
-            let ch = self.read_char();
-            match ch {
-                Some(ch) => {
-                    let tok = match ch {
-                        '{' => FuncToken::LeftBrace,
-                        '}' => FuncToken::RightBrace,
-                        '(' => FuncToken::LeftParen,
-                        ')' => FuncToken::RightParen,
-                        ',' => FuncToken::Comma,
-                        '"' => self.read_string(),
-                        '#' => self.read_ifdef(),
-                        ch if ch.is_numeric() => self.read_number(),
-                        ch if ch.is_alphabetic() => self.read_identifier(),
-                        ch if ch.is_whitespace() => continue,
-                        ';' => continue,
-                        '/' => {
-                            self.skip_line();
-                            continue;
-                        }
-                        ch => todo!("unhandled: {:?}", ch),
-                    };
-
-                    result.push(tok);
-                }
-                None => break,
+        while let Some(ch) = self.chars.next() {
+            let tok = match ch {
+                '{' => self.process_left_brace(),
```

Comment:
I don't think there is any particular reason we need to make these all functions if they just return a value, i think it muddies the code

Reply:
It's true that these functions just return a value. 

I did it for consistency and decoupling. If the logic of process_left_brace should change in the future, it's easier for another contributor to say with confidence that the logic is probably contained in the struct method.

Of course it's debatable that inlining the logic is also obvious that the logic is contained there. 

I can revert back if you feel strongly about it